front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/index.ts

```diff
@@ -24,8 +25,35 @@ async function handler(
     return apiError(req, res, {
       status_code: 404,
       api_error: {
-        type: "user_not_found",
-        message: "Could not find the user.",
+        type: "data_source_not_found",
+        message: "The data source you requested was not found.",
+      },
+    });
+  }
+
+  const { dsId } = req.query;
+  if (typeof dsId !== "string") {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid path parameters.",
+      },
+    });
+  }
+
+  const dataSource = await DataSourceResource.fetchByNameOrId(
+    auth,
+    dsId,
+    // TODO(DATASOURCE_SID): Clean-up
```

Comment:
nit: what cleanup would be needed here? change the call to "fetchById" when we remove "name"? IMO we can remove the com (will be done auto by IDE renaming)